pytest_reportportal/service.py

```diff
@@ -231,6 +266,79 @@ def _stop_if_necessary(self):
         except queue.Empty:
             pass
 
+
+    @staticmethod
+    def _add_item_hier_parts_dirs(item, hier_flag, dirs_level, report_parts, dirs_parts, rp_name=""):
+
+        parts_dirs = PyTestServiceClass._get_item_dirs(item)
+        dir_path = item.fspath.new(dirname="", basename="", drive="")
+        for dir_name in parts_dirs[dirs_level:]:
+            dir_path = dir_path.join(dir_name)
+            path = str(dir_path)
+
+            if hier_flag:
+                if path in dirs_parts:
+                    item_dir = dirs_parts[path]
+                else:
+                    item_dir = File(dir_name, nodeid=dir_name, session=item.session, config=item.session.config)
+                    item_dir._rp_name = dir_name
+                    dirs_parts[path] = item_dir
+
+                report_parts.append(item_dir)
+            else:
+                rp_name = path
+
+        return rp_name
+
+    @staticmethod
+    def _add_item_hier_parts_parametrize(item, report_parts, tests_parts, rp_name=""):
+
+        for mark in item.own_markers:
+            if mark.name == 'parametrize':
+                ch_index = item.nodeid.find("[")
+                test_fullname = item.nodeid[:ch_index if ch_index > 0 else len(item.nodeid)]
+                test_name = item.originalname
+
+                rp_name += ("::" if rp_name else "") + test_name
+
+                if test_fullname in tests_parts:
+                    item_test = tests_parts[test_fullname]
+                else:
+                    item_test = Item(test_fullname, nodeid=test_fullname, session=item.session, config=item.session.config)
+                    item_test._rp_name = rp_name
+                    item_test.obj = item.obj
+                    item_test.keywords = item.keywords
+                    item_test.own_markers = item.own_markers
+                    item_test.parent = item.parent
+
+                    tests_parts[test_fullname] = item_test
+
+                rp_name = ""
+                report_parts.append(item_test)
+                break
+
+        return rp_name
+
+    @staticmethod
+    def _add_item_hier_parts_other(item_parts, item, type, hier_flag, report_parts, rp_name=""):
+
+        for part in item_parts:
+
+            if isinstance(part, type):
+
+                if type is Module:
+                    module_path = str(item.fspath.new(dirname=rp_name, basename=part.fspath.basename, drive=""))
+                    rp_name = module_path[1:]
+                elif type in (Class, Function):
+                    rp_name += ("::" if rp_name else "") + part.name
+
+                if hier_flag:
+                    part._rp_name = rp_name
+                    rp_name = ""
+                    report_parts.append(part)
+
+        return rp_name
```

Comment:
I do not understand: why do we have rp_name?
we receive rp_name as argument, then update it in the loop and return only last value. correct?

Reply:
See previous comments